### PR TITLE
Created a Community Page

### PR DESCRIPTION
**Pull request by elskies**
You can reach the page by "/community", I did not add a link in navbar

### CODE REVIEW

**Comment by vck3000** [comment]
image

Nationality is defaulted to null and doesn't appear if the user has never updated their profile/biography! Please fix this case. 

Also, players with no custom avatar don't show up with the normal base res/spy images.

Also, not sure if intentional, but biography text isn't vertically aligned (whereas nationality is). 

Finally, I'd also add in some limiting number of players to be shown. We have 4000 registered users, and 223 users with more than 49 games played. So it'll fill up the screen very quickly! 

1) You can still render all 223 players (since its not THAT much) and have a "load more" button to slowly reveal more. Note that this will be unsorted though? By default I reckon it'd sort based on when the user file was created (so in order of who joined the site first).

OR

2) You can do something like my forums page where theres a `/<pagenumber>` at the end of the URL and feed in those people there. 

If you do number 2), then you'll need a way to sort players. You can do this through most games played in last 30 days?? (its not being recorded right now though so you'd have to code that in.) Or you can do it for most recent game played? (again, not being recorded, but much easier to implement).

File: routes/index.js [hunk 1]
```diff
@@ -20,6 +20,36 @@ const request = require('request');
...
+		        	if(err) {
+		            	console.log(err);
+		          	}
+		          	else {
+		            	res.render("community", {users:allUsers, mods:allMods, currentUser:req.user});
+          			}
+	        	});
+        	}
+	// sort by games played
+	}).sort({totalGamesPlayed: -1});
```
[the remark this answers is not in the document]
Comment (resskies) [reply routes/index.js]: yes it sorts by total games played

[the remark this answers is not in the document]
Comment (vck3000) [reply routes/index.js]: But it's run after `res.render()` is called...?

[the remark this answers is not in the document]
Comment (resskies) [reply routes/index.js]: Well I tested it. You can un-comment the #games column in community.ejs table to see the sorting.

**Comment by resskies** [comment]
Fixed nationality, biography and show more set to 50 users only

**Comment by vck3000** [comment]
Sorry I haven't checked back on this recently. I'll get to it tonight/tmr night.

**Comment by vck3000** [comment]
Big server error when trying to load /community page.

```
2019-01-30T12:11:51.393913+00:00 app[web.1]: ReferenceError: /app/views/community.ejs:164
2019-01-30T12:11:51.393932+00:00 app[web.1]:     162|                     <td><img src="https://www.proavalon.com/avatars/base-spy.png"></td>                
2019-01-30T12:11:51.393933+00:00 app[web.1]:     163|                 <% } else { %>
2019-01-30T12:11:51.393935+00:00 app[web.1]:  >> 164|                     <td><img src="<%= user.avatarImgSpy %>"></td>
2019-01-30T12:11:51.393936+00:00 app[web.1]:     165|                 <% } %>
2019-01-30T12:11:51.393938+00:00 app[web.1]:     166|                 <%if (mod.nationality === "") { %>
2019-01-30T12:11:51.393940+00:00 app[web.1]:     167|                     <td class="nat">United Nations</td>
2019-01-30T12:11:51.393942+00:00 app[web.1]: 
2019-01-30T12:11:51.393943+00:00 app[web.1]: user is not defined
2019-01-30T12:11:51.393945+00:00 app[web.1]:     at eval (eval at compile (/app/node_modules/ejs/lib/ejs.js:618:12), <anonymous>:255:26)
2019-01-30T12:11:51.393946+00:00 app[web.1]:     at Array.forEach (<anonymous>)
2019-01-30T12:11:51.393947+00:00 app[web.1]:     at eval (eval at compile (/app/node_modules/ejs/lib/ejs.js:618:12), <anonymous>:231:13)
2019-01-30T12:11:51.393949+00:00 app[web.1]:     at returnedFn (/app/node_modules/ejs/lib/ejs.js:653:17)
2019-01-30T12:11:51.393950+00:00 app[web.1]:     at tryHandleCache (/app/node_modules/ejs/lib/ejs.js:251:36)
2019-01-30T12:11:51.393951+00:00 app[web.1]:     at View.exports.renderFile [as engine] (/app/node_modules/ejs/lib/ejs.js:482:10)
2019-01-30T12:11:51.393952+00:00 app[web.1]:     at View.render (/app/node_modules/express/lib/view.js:135:8)
2019-01-30T12:11:51.393953+00:00 app[web.1]:     at tryRender (/app/node_modules/express/lib/application.js:640:10)
2019-01-30T12:11:51.393955+00:00 app[web.1]:     at Function.render (/app/node_modules/express/lib/application.js:592:3)
2019-01-30T12:11:51.393956+00:00 app[web.1]:     at ServerResponse.render (/app/node_modules/express/lib/response.js:1008:7)
2019-01-30T12:11:51.393957+00:00 app[web.1]:     at /app/routes/index.js:46:20
2019-01-30T12:11:51.393958+00:00 app[web.1]:     at /app/node_modules/mongoose/lib/model.js:4669:16
2019-01-30T12:11:51.393959+00:00 app[web.1]:     at /app/node_modules/mongoose/lib/query.js:4090:12
2019-01-30T12:11:51.393961+00:00 app[web.1]:     at process.nextTick 
```

File: views/community.ejs [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+                <td><%= mod.username %></td>
+                <%if (mod.avatarImgRes === null) { %>
+                    <td><img src="https://www.proavalon.com/avatars/base-res.png"></td>
+                <% } else { %>
+                    <td><img src="<%= mod.avatarImgRes %>"></td>
+                <% } %>
+                <%if (mod.avatarImgSpy === null) { %>
+                    <td><img src="https://www.proavalon.com/avatars/base-spy.png"></td>                
+                <% } else { %>
+                    <td><img src="<%= user.avatarImgSpy %>"></td>
```
Comment (vck3000) [inline views/community.ejs]: This should probably say `mod.avatarImgSpy` instead of `user.avatarImgSpy`...

Comment (elskies) [reply views/community.ejs]: Yeah you're right, I fixed it

**Pull request merged**